### ISSUE TITLE
Canvas become transparent when it was shown from hidden state

### ISSUE THREAD

**Issue by hououinkami** [Hammerspoon/hammerspoon#2809]
I create some canvas objects and show them, then I used hide method to hide them. 
When I show them by show method, these canvas become transparent just like the alpha value changed. 
But if I used delete method instead of hide, it works well.
Is this a bug？
Thanks very much!

**Comment by latenitefilms** [comment]
Ummm, this test seems to work fine?

```lua
c = require("hs.canvas")
a = c.new{x = 100, y = 100, w = 200, h = 200 }:show()
a[1] = {
  type="image",
  -- the Hammerspoon menu icon has this name within the Application bundle, so we can get it this way:
  image = hs.image.imageFromName("statusicon"):template(false),
  imageScaling = "scaleToFit"
}
a[2] = {
  type = "rectangle",
  action = "fill",
  fillGradientColors = {
    { red = 1 },
    { red = 1, green = .65 },
    { red = 1, green = 1 },
    { green = 1 },
    { blue = 1 },
    { red = .30, blue = .5 },
    { red = .93, green = .5, blue = .93 }
  },
  fillGradient = "radial"
}
  
hs.timer.doAfter(2, function() a:hide() end)
hs.timer.doAfter(4, function() a:show() end)
hs.timer.doAfter(6, function() a:delete() end)
```

**Comment by hououinkami** [comment]
> Ummm, this test seems to work fine?
> 
> ```lua
> c = require("hs.canvas")
> a = c.new{x = 100, y = 100, w = 200, h = 200 }:show()
> a[1] = {
>   type="image",
>   -- the Hammerspoon menu icon has this name within the Application bundle, so we can get it this way:
>   image = hs.image.imageFromName("statusicon"):template(false),
>   imageScaling = "scaleToFit"
> }
> a[2] = {
>   type = "rectangle",
>   action = "fill",
>   fillGradientColors = {
>     { red = 1 },
>     { red = 1, green = .65 },
>     { red = 1, green = 1 },
>     { green = 1 },
>     { blue = 1 },
>     { red = .30, blue = .5 },
>     { red = .93, green = .5, blue = .93 }
>   },
>   fillGradient = "radial"
> }
>   
> hs.timer.doAfter(2, function() a:hide() end)
> hs.timer.doAfter(4, function() a:show() end)
> hs.timer.doAfter(6, function() a:delete() end)
> ```

Thanks for your reply!
I replace your example code into mine (function togglecanvas()) as follows, it works well, so I think the problem may be come from my code which build the canvas.
`c = require("hs.canvas")
function setCanvas()
    a = c.new{x = 100, y = 100, w = 200, h = 200 }
    a[1] = {
        type="image",
        -- the Hammerspoon menu icon has this name within the Application bundle, so we can get it this way:
        image = hs.image.imageFromName("statusicon"):template(false),
        imageScaling = "scaleToFit"
    }
    a[2] = {
        type = "rectangle",
        action = "fill",
        fillGradientColors = {
            { red = 1 },
            { red = 1, green = .65 },
            { red = 1, green = 1 },
            { green = 1 },
            { blue = 1 },
            { red = .30, blue = .5 },
            { red = .93, green = .5, blue = .93 }
        },
    fillGradient = "radial"
    }
end
function togglecanvas()
    if a:isShowing() == true then
        a:hide()
    else
        a:show()
    end
end
setCanvas()
testBar = hs.menubar.new()
testBar:setTitle('TEST')
testBar:setClickCallback(togglecanvas)`
If I use my code, this problem will appear randomly, especially when the togglecanvas function is triggered frequently.
It has many elements so I can't find out which code snippet is wrong

**Comment by latenitefilms** [comment]
Yes, it looks like things break if you try and hide or show it when it's already in the process of fading to hide or show.

You could add some logic to work around it, for example:

```lua
c = require("hs.canvas")
function setCanvas()
    a = c.new{x = 100, y = 100, w = 200, h = 200 }
    a[1] = {
        type="image",
        -- the Hammerspoon menu icon has this name within the Application bundle, so we can get it this way:
        image = hs.image.imageFromName("statusicon"):template(false),
        imageScaling = "scaleToFit"
    }
    a[2] = {
        type = "rectangle",
        action = "fill",
        fillGradientColors = {
            { red = 1 },
            { red = 1, green = .65 },
            { red = 1, green = 1 },
            { green = 1 },
            { blue = 1 },
            { red = .30, blue = .5 },
            { red = .93, green = .5, blue = .93 }
        },
    fillGradient = "radial"
    }
end

local fadeTime = 0.6
local isFading = false

function togglecanvas()
	if isFading then
		print("already fading")
		return
	end
	
	isFading = true
	
    if a:isShowing() == true then	    
        a:hide(fadeTime)
    else    	
        a:show(fadeTime)
    end
    
    hs.timer.doAfter(fadeTime, function() isFading = false end)
end

setCanvas()
testBar = hs.menubar.new()
testBar:setTitle('TEST')
testBar:setClickCallback(togglecanvas)
```

The above code is currently just "ignoring" the clicks if a fade is already in action, but you could add another timer to delay the show/hide until after the first one is complete.

@asmagill - Would you consider this a bug?

**Comment by hououinkami** [comment]
Use timer to delay the fade action works better, thanks very much!
I test this code and the problem occurs randomly at a lower frequency.
